webhonc/src/index.ts

```diff
@@ -0,0 +1,94 @@
+import { zValidator } from "@hono/zod-validator";
+import { Hono } from "hono";
+import { z } from "zod";
+import type { Bindings, Variables } from "./types";
+import { WebHonc } from "./webhonc";
+
+const app = new Hono<{ Bindings: Bindings; Variables: Variables }>();
+
+app.get("/ws", async (c) => {
+  if (c.req.header("upgrade") !== "websocket") {
+    return new Response("Not a websocket request", { status: 426 });
+  }
+
+  const id = c.env.WEBHONC.newUniqueId();
+  const webhonc = c.env.WEBHONC.get(id) as DurableObjectStub<WebHonc>;
+
+  return webhonc.fetch(c.req.raw);
+});
+
+app.all(
+  "/:id/*",
+  zValidator("param", z.object({ id: z.string() })),
+  async (c) => {
+    const { id } = c.req.valid("param");
+
+    // We need to serialize the path components without the id
+    // so we can send it down the wire and replay it locally
+    const pathComponentsWithoutId = c.req.path
+      .split("/")
+      .filter((comp) => comp.length > 0)
+      .slice(1);
+
+    const method = c.req.method;
+
+    const contentType = c.req.header("content-type");
+
+    let body: string | FormData | null;
+
+    // if the request is a GET or HEAD, we don't want to send the body
+    if (method === "GET" || method === "HEAD") {
+      body = null;
+    } else {
+      switch (contentType) {
+        case "application/json":
+          body = await c.req.json();
+          break;
+        case "application/x-www-form-urlencoded":
+          body = await c.req.formData();
+          break;
+        case "text/plain":
+          body = await c.req.text();
+          break;
+        default:
+          body = null;
+          break;
+      }
+    }
+
+    const query = c.req.query();
+
+    const doId = c.env.WEBHONC.idFromString(id);
+    const webhonc = c.env.WEBHONC.get(doId) as DurableObjectStub<WebHonc>;
+
+    const headers = c.req.raw.headers;
+    const headersJson: { [key: string]: string } = {};
+    for (const [key, value] of headers.entries()) {
+      // these are calculated dynamically by the fetch client so we don't send them down the wire
+      if (
+        key.toLowerCase() === "content-length" ||
+        key.toLowerCase() === "content-type"
```

Comment:
wait i don't think you want to abandon the `content-type` here? unless i'm missing something. 

oddly when i set application/json as an explicit header to webhonc, i see this in the UI for the request that's ultimately proxied (`application/json, application/json`)

<img width="691" alt="image" src="https://github.com/user-attachments/assets/0b7e92da-7548-42e8-9f8e-c453df55a443">

Reply:
to be clear, you **SHOULD** abandon the header for multipart form data, since the form data boundary is generated dynamically afaik, but we're not handling multipart requests here anyhow

Reply:
yea not sure why I decided to wipe `Content-Type` - changing this